presto-client/src/main/java/io/prestosql/client/auth/external/TokenPoll.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.client.auth.external;
+
+import java.util.Optional;
+
+import static io.prestosql.client.auth.external.TokenPoll.Status.FAILED;
+import static io.prestosql.client.auth.external.TokenPoll.Status.PENDING;
+import static io.prestosql.client.auth.external.TokenPoll.Status.SUCCESSFUL;
+import static java.util.Objects.requireNonNull;
+
+class TokenPoll
```

Comment:
I think this class is not very useful. It's not really good at encapsulating different states of the process and rules which one has to follow as one can call `getNextTokenUri` for a failed token or `getError` for a successful one.